docs/csharp/how-to/index.md

```diff
@@ -0,0 +1,241 @@
+- [Serialize using <xref:System.Xml.Serialization.XmlSerializer>](../programming-guide/concepts/linq/how-to-serialize-using-xmlserializer.md).
+- [Use annotations to transform XML trees into object graphs](../programming-guide/concepts/linq/how-to-use-annotations-to-transform-linq-to-xml-trees-in-an-xslt-style.md).
+- [Stream XML fragments including header information](../programming-guide/concepts/linq/how-to-stream-xml-fragments-with-access-to-header-information.md).
+- [Stream XML fragments](../programming-guide/concepts/linq/how-to-stream-xml-fragments-from-an-xmlreader.md).
+- [Query XML using XPath](../programming-guide/concepts/linq/how-to-query-linq-to-xml-using-xpath.md).
+- [Populate an XML tree from the file system](../programming-guide/concepts/linq/how-to-populate-an-xml-tree-from-the-file-system.md).
+- [Create XML from CSV file content](../programming-guide/concepts/linq/how-to-generate-xml-from-csv-files.md).
+- [Project XML data into an object graph](../programming-guide/concepts/linq/how-to-project-an-object-graph.md).
+- [Find the root element using XPath](../programming-guide/concepts/linq/how-to-find-the-root-element-xpath-linq-to-xml.md).
+- [Transform the shape of an XML tree](../programming-guide/concepts/linq/how-to-transform-the-shape-of-an-xml-tree.md).
+- [Generate text files from XML](../programming-guide/concepts/linq/how-to-generate-text-files-from-xml.md).
+- [Transform large XML documents from an input stream](../programming-guide/concepts/linq/how-to-perform-streaming-transform-of-large-xml-documents.md).
+- [Perform streaming transformations of text to XML](../programming-guide/concepts/linq/how-to-perform-streaming-transformations-of-text-to-xml.md).
+- [Work with dictionaries using LINQ to XML](../programming-guide/concepts/linq/how-to-work-with-dictionaries-using-linq-to-xml.md).
+- [Retrieve paragraphs from an office open XML document](../programming-guide/concepts/linq/how-to-retrieve-paragraphs-from-an-office-open-xml-document.md).
+- [Modify an office Open XML document](../programming-guide/concepts/linq/how-to-modify-an-office-open-xml-document.md).
+
+## Multiple threads and async processing
+
+Modern programs often use asynchronous operations. These articles will help you learn
+to use these techniques.
+
+- [Improve async performance using <xref:System.Threading.Tasks.Task.WhenAll%2A>](../programming-guide/concepts/async/how-to-extend-the-async-walkthrough-by-using-task-whenall.md).
+- [Make multiple web requests in parallel using `async` and `await`](../programming-guide/concepts/async/how-to-make-multiple-web-requests-in-parallel-by-using-async-and-await.md).
+- [Use a thread pool](../programming-guide/concepts/threading/how-to-use-a-thread-pool.md).
+
+## Command line args to your program
+
+Typically, C# programs have command line arguments. These articles teach you to access and process
+those command line arguments.
```

Comment:
Although both of the topics are about displaying command line arguments, I'd change the description here to retrieving them, which strikes me as a much more common operation than displaying them.